tests/test_modutils.py

```diff
@@ -335,6 +335,22 @@ def test_pyi_preferred(self) -> None:
             os.path.normpath(module) + "i",
         )
 
+    def test_nonstandard_extension(self) -> None:
+        package = resources.find("pyi_data/find_test")
+        modules = [
+            os.path.join(package, "__init__.weird_ext"),
+            os.path.join(package, "standalone_file.weird_ext"),
```

Comment:
```suggestion
            os.path.join(package, "__init__.weird_ext"),
            os.path.join(package, "standalone_file.weird_ext"),
            os.path.join(package, "standalone_file.py"),
```
I might have reviewed this still half asleep the first time* but shouldn't it be this instead (issue talks about 'two file with the same name but different extension') ? I suppose being able to also handle 'two file with the same extension but different name is also valuable.

* (just found and read the pylint issue to close it automatically when we merge this and it doesn't match)

Reply:
There already exists a __init__.py, so I figured I could just reuse that file for my tests too. And I also threw in a standalone file without any weird naming and without any other file sharing it's base name just because  ¯\_(ツ)_/¯

Reply:
Should we perhaps test for `__init__.py` too? I just realized there doesn't seem to be a test for that. Most likely other test would catch that error though.

Reply:
It's okay let's test once